src/fsharp/FSharp.Core/map.fs

```diff
@@ -64,33 +51,30 @@ namespace Microsoft.FSharp.Collections
 
         let empty = MapEmpty 
 
-        let height  = function
-          | MapEmpty -> 0
-          | MapOne _ -> 1
-          | MapNode(_,_,_,_,h) -> h
+        let size x =
+            match x with
+            | MapEmpty -> 0
+            | MapNode (_,_,_,_,h) -> h
```

Comment:
We should rename h here to size

Reply:
Also make it inline

Reply:
Currently all this is hanging off the end of #5307. Maybe I should rebranch off current master?

Reply:
@manofstick if possible, this would be best, yes. We can evaluate this separately which makes it easier to review, test, and incorporate.

Reply:
@cartemp

Happy to do so, if you are happy to trade memory for performance (which is what the removal of MapOne means, as per PRs text). I mean I'm happy to spend time doing this as long as it's not completely my wasted time...

Reply:
Hmmm, we can think about it. From the standpoint of making it easier to merge, having this be entirely separate is best, but if doing so is too difficult (or at least wouldn't make much sense without the others) then I think it's fine to keep working as-is.

Since #5307 is marked as approved for F# vNext, I've tagged it with the dev16.0 milestone, this could just as well stay as it is today and be considered in the context of #5307. It's just that our evaluation of it will also depend on that, and it is generally less likely for something to come in if it takes a dependency on something else coming in. But I think that risk is smaller given that the dependent branch is on the path towards approval.